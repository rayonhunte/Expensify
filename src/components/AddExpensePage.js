import React from 'react';


const AddExpensePage = () => (
  <div>
    this is from add expensify
  </div>
);


export default AddExpensePage;