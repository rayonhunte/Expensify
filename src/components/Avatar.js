import React from 'react';
import {connect}  from 'react-redux';
import {Link} from 'react-router-dom';

const Avatar = ({profile, height, width})=>(
  <div>
    <Link to="/profile">
      <img className="avatar" src={profile.photoURL} height={height} width={width} border="5"/>
    </Link>
  </div>
);

const mapStateToProps = (state) =>{
  return {
    profile: state.profile
  };
};
  
export default connect(mapStateToProps)(Avatar);