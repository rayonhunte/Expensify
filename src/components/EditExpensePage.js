import React from 'react';


const EditExpensePage = (props) => {
  console.log(props);
  return (
    <div>
      Editing {props.match.params.id}
    </div>
  )
};


export default EditExpensePage;