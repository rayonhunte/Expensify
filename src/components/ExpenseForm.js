import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';


export default class ExpenseForm extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      description: props.expense ? props.expense.description:'',
      note: props.expense ? props.expense.note:'',
      amount: props.expense ? (props.expense.amount / 100).toString():'',
      createAt: props.expense ? moment(props.expense.createAt):moment(),
      calendarFocused: false,
      error: ''
    }
  }
  onDescriptionChange = (e) =>{
    const description = e.target.value;
    this.setState(()=>({
      description
    }))
  };
  OnNoteChange = (e) =>{
    const note = e.target.value;
    this.setState(()=>({
      note
    }))
  }
  onAmountChange = (e) =>{
    const amount = e.target.value;
    if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
      this.setState(()=>({amount}))
    }
  }

  onDateChange = (createdAt)=>{
    if(createdAt){
      this.setState(()=>({
        createdAt
      }))
    }
    
  }

  onFocusChange = ({focused})=>{
    this.setState(()=>({
      calendarFocused: focused
    }))
  }  

  onSubmit =(e)=>{
    e.preventDefault();
    if(!this.state.description || !this.state.amount){
        this.setState(()=>({
          error: "this was an error"
        }))
    }else{
        this.setState(()=>({error:''}))
        this.props.onSubmit({
          description: this.state.description,
          amount: parseFloat(this.state.amount, 10) * 100,
          createAt: this.state.createAt.valueOf(),
          note: this.state.note
        })
    }
  }

  render(){
    return (
      <div>
        <div>
          {this.state.error && <p>{this.state.error}</p>}
        </div>
        <form onSubmit={this.onSubmit}>
          <input type="text"
           placeholder="Description"
           autoFocus  value={this.state.description}
           onChange={this.onDescriptionChange}
           />
          <input type="text" 
           placeholder="Amount" 
           value={this.state.amount}
           onChange={this.onAmountChange}
           />
           <SingleDatePicker 
             date ={this.state.createAt}
             onDateChange ={this.onDateChange}
             focused = {this.state.calendarFocused}
             onFocusChange = {this.onFocusChange}
             numberOfMonths={1}
             isOutsideRange={()=>(false)}
           />
          <textarea placeholder="Add a note optional" 
          value={this.state.note}
          onChange={this.OnNoteChange}
          >      
          </textarea>
          <button type="submit"> Add Expense</button>
        </form>
      </div>
    )
  }
}

