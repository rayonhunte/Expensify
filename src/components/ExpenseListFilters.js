import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';


class ExpenseListFilters extends React.Component{
  state ={
    calendarFocused: null
  }
  onDatesChange = ({startDate, endDate})=>{
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
    // console.log(endDate)
    // if (startDate){
    //   this.props.dispatch(setStartDate(startDate));
    // }
    // if (endDate){
    //   this.props.dispatch(setEndDate(endDate));
    // }
    
  }
  onFocusChange = (calendarFocused)=>{
    this.setState(()=>({
      calendarFocused
    }
    ))
  }
  render(){
    return (
      <div>
      <input type="text" value={this.props.filters.text} onChange={(e)=>{
        this.props.dispatch(setTextFilter(e.target.value))
      }}/>
      <select value = {this.props.filters.sortBy} 
      onChange={(e)=>{
        if(e.target.value !== 'date'){
          this.props.dispatch(sortByAmount())
        }else{
          this.props.dispatch(sortByDate())
        }
      }}>
        <option value='date'>Date</option>
        <option value='amount'>Amount</option>
      </select>
      <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
    </div>
    );
  }
}

const mapStateProps = (state) =>{
  return {
    filters: state.filters
  }
}

export default connect(mapStateProps)(ExpenseListFilters);