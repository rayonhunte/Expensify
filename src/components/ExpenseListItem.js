import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


const ExpensesListItem = ({dispatch, description ,id, amount, createAt})=>(
    <div>
      <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
      </Link>
      <p>
      {numeral(amount / 100).format('$0.0.00')} 
      {moment(createAt).format('MMMM Do, YYYY' )}
      </p>
    </div>
);

export default ExpensesListItem;

//test comment