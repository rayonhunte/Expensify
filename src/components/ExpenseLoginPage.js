import React from 'react';
import {connect} from 'react-redux';
import { startLogin } from '../actions/auth';

export const  ExpenseLoginPage = ({startLogin})=> {
  return (
      <div className="box-layout">
          <div className="box-layout__box">
            <h1 className="layout__title">React Expensify</h1>
            <p>Get your expenses in Order !!!</p>
            <button onClick={startLogin} className="button-layout">Login With Google</button>
          </div>
      </div>
    );
  
};

const mapDispatchToProps = (dispatch)=>({
  startLogin: ()=> dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(ExpenseLoginPage);