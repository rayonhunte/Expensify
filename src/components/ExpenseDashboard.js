import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboard = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboard;