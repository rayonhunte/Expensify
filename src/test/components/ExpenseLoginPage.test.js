import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseLoginPage} from '../../components/ExpenseLoginPage';


let startLogin, wrapper;


beforeEach(()=>{
  startLogin = jest.fn();
  wrapper = shallow(<ExpenseLoginPage startLogin={startLogin}/>);
});


test('should render login', ()=>{
  expect(wrapper).toMatchSnapshot();
});





test('test login',()=>{
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalledWith();  
});