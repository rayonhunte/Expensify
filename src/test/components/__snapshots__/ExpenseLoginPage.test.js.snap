// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="layout__title"
    >
      React Expensify
    </h1>
    <p>
      Get your expenses in Order !!!
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login With Google
    </button>
  </div>
</div>
`;
